src/main/scala/gestalt/Trees.scala

```diff
@@ -83,6 +89,13 @@ trait Trees extends Params with TypeParams with
   // modifiers
   def emptyMods: Mods
 
+  // typed trees
+  val tpd: TypedTrees
+  trait TypedTrees {
+    type Tree      >: Null <: AnyRef
+    type Param     <: Tree
+    type ValDef    <: Tree
+  }
```

Comment:
`ValDef` and `Param` are not actually used. It seems for typed trees, it's fine to just provide a single type `Tree` or another `TypeTree`. As typed tree construction has so many constraints, the type guarantees a high degree of safety by construction -- the type of trees only provide a very low-level safety far from enough for typed trees.